src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/LeftAlignAndTrimVariantsUnitTest.java

```diff
@@ -0,0 +1,107 @@
+package org.broadinstitute.hellbender.tools.walkers.variantutils;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.variant.variantcontext.VariantContextBuilder;
+import org.broadinstitute.hellbender.GATKBaseTest;
+import org.broadinstitute.hellbender.engine.ReferenceMemorySource;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.variantcontext.Allele;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.UnsupportedEncodingException;
+import java.util.*;
+import java.util.stream.*;
+
+public class LeftAlignAndTrimVariantsUnitTest extends GATKBaseTest {
+    final String refBases1 = "ACAGAGCTGACCCTCCCTCCCCTCTCCCAGTGCAACAGCACGGGCGGCGACTGCTTTTACCGAGGCTACACGTCAGGCGTGGCGGCTGTCCAGGACTGGTACCACTTCCACTATGTGGATCTCTGCTGAGGACCAGGAAAGCCAGCACCCGCAGAGACTCTTCCCCAGTGCTCCATACGATCACCATTCTCTGCAGAAGG";
+    final List<String> longPieces = Arrays.asList("AAAAAAAAAAAAAAAAAAAAAAAAAAAA", "TCTCTCTCTCTCTC", "CAGTCCAGTCCAGTCCAGTCCAGTCCAGTCCAGTCCAGTCCAGTCCAGTCCAGTC", Utils.dupString("A", 120)); // where we'll perform tests
+    final List<Integer> strLengths = Arrays.asList(1, 2, 5, 1); //
+    final List<String> refBasesStrings = longPieces.stream().map(l -> refBases1 + l + refBases1).collect(Collectors.toList());
+
+    final List<Integer> contigStops = refBasesStrings.stream().map(String::length).collect(Collectors.toList());
+    final List<SAMFileHeader> headers = contigStops.stream().map(c ->
+            ArtificialReadUtils.createArtificialSamHeader(1, 1, c)).collect(Collectors.toList());
+    final String artificialContig = "1";
+    final int repeatStart = refBases1.length();
+    final List<SimpleInterval> refIntervals = contigStops.stream().map(c ->
+            new SimpleInterval(artificialContig, 1, c)).collect(Collectors.toList());
+    final List<ReferenceBases> refBases = IntStream.range(0, longPieces.size()).mapToObj(i -> new ReferenceBases(refBasesStrings.get(i).getBytes(), refIntervals.get(i)))
+            .collect(Collectors.toList());
+    //final ReferenceMemorySource refSource=new ReferenceMemorySource(refBases,header.getSequenceDictionary());
+    final List<ReferenceMemorySource> refSources = IntStream.range(0, longPieces.size()).mapToObj(i -> new ReferenceMemorySource(refBases.get(i), headers.get(i).getSequenceDictionary()))
+            .collect(Collectors.toList());
+
+    @DataProvider(name = "LeftAlignDataProvider")
+    public Object[][] LeftAlignTestData() throws UnsupportedEncodingException {
+        List<Object[]> tests = new ArrayList<Object[]>();
```

Comment:
This data provider needs to be better formatted with more comments to explain exactly what it is generating as test data.

Reply:
You should probably create a second data provider for this test that test larger repeat units rather than repeating the same alt-base over and over again.